### PR TITLE
Install psycopg2 module

### LINKED ISSUE #2686: psycopg2 wheel missing from Docker image
**Describe the bug**
Argilla cannot start with a PostgreSQL database backend, as the psycopg2 wheel is missing from the Docker image.

**To Reproduce**
1. Start up an Argilla 1.6.0 container defining `ARGILLA_DATABASE_URL` with a PostgreSQL URL.

**Expected behavior**
Argilla uses the PostgreSQL database backend.

**Screenshots**
`ModuleNotFoundError: No module named 'psycopg2'`

**Environment (please complete the following information):**
 - OS [e.g. iOS]: Linux
 - Browser [e.g. chrome, safari]: N/A
 - Argilla Version [e.g. 1.0.0]: 1.6.0
 - ElasticSearch Version [e.g. 7.10.2]: 8.5.3
 - Docker Image (optional) [e.g. argilla:v1.0.0]: argilla:v1.6.0

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by frascuchon**
# Description

In this PR, the official docker image is built including the psycopg2 module, which enables image compatibility with PostgreSQL DB.

Closes #2686

**Type of change**

(Please delete options that are not relevant. Remember to title the PR according to the type of change)

- [x] Improvement (change adding some improvement to an existing functionality)

**How Has This Been Tested**

Only local tests are done.

**Checklist**

- [x] I have merged the original branch into my forked branch
- [ ] I added relevant documentation
- [x] follows the style guidelines of this project
- [x] I did a self-review of my code
- [ ] I made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by dbuades** [comment]
Thank you for this! We no longer need to build our own Docker image 💪

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,6 +19,12 @@ These are the section headers that we use:
 
 ### Added
 
+- Argilla server docker image is built with PostgreSQL support. Closes [#2686](https://github.com/argilla-io/argilla/issues/2686)
```
Comment (tomaarsen) [inline CHANGELOG.md:22]: Perhaps this should be under "Changed".

So this won't be included in 1.7.0, right?

Comment (frascuchon) [reply CHANGELOG.md:22]: Yes, better as "Changed".

I don't want to block the next release, but if this is working soon, I think we can include it in 1.7.0

**Review by tomaarsen — APPROVED** [review]
Seems to align with the [build prerequisites](https://www.psycopg.org/docs/install.html#build-prerequisites). I don't have PostgreSQL locally, so I can't really test this locally. That said, the updated snippet (i.e. update, install reqs, install psycopg2, uninstall reqs) does run for me locally without issues.

**Comment by frascuchon** [comment]
You can use the built image with the following docker-compose yaml configuration (I've been doing some tests and works as expected)

```yaml
version: "3.9"

services:

  db:
    image: postgres:14.1-alpine
    restart: unless-stopped
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
    ports:
      - "5432:5432"
    networks:
      - argilla
    volumes:
      - dbdata:/var/lib/postgresql/datas
  argilla:
    image: argilla/argilla-server:pr-2820
    restart: unless-stopped
    ports:
      - "6900:6900"
    environment:
      ARGILLA_HOME_PATH: /var/lib/argilla
      ARGILLA_DATABASE_URL: postgresql+psycopg2://postgres:postgres@db:5432/
      ARGILLA_ELASTICSEARCH: http://elasticsearch:9200
      # ARGILLA_ENABLE_TELEMETRY: 0 # Opt-out for telemetry https://docs.argilla.io/en/latest/reference/telemetry.html

      # Set user configuration https://docs.argilla.io/en/latest/getting_started/installation/user_management.html
      # ARGILLA_LOCAL_AUTH_USERS_DB_FILE: /config/.users.yaml
      # volumes:
      #- ${PWD}/.users.yaml:/config/.users.yaml

      # DEFAULT_USER_ENABLED: false # Uncomment this line to disable the creation of the default user
      # DEFAULT_USER_PASSWORD: custom-password # Uncomment this line to set a custom password for the default user
      # DEFAULT_USER_API_KEY: custom-api-key # Uncomment this line to set a custom api-key for the default user
    networks:
      - argilla
    volumes:
      # ARGILLA_HOME_PATH is used to define where Argilla will save it's application data.
      # If you change ARGILLA_HOME_PATH value please copy that same value to argilladata volume too.
      - argilladata:/var/lib/argilla
  elasticsearch:
    image: docker.elastic.co/elasticsearch/elasticsearch:8.5.3
    environment:
      - node.name=elasticsearch
      - cluster.name=es-argilla-local
      - discovery.type=single-node
      - "ES_JAVA_OPTS=-Xms512m -Xmx512m"
      - cluster.routing.allocation.disk.threshold_enabled=false
      - xpack.security.enabled=false
    ulimits:
      memlock:
        soft: -1
        hard: -1
    networks:
      - argilla
    ports:
      - "9200:9200"
      - "9300:9300"
    volumes:
      - elasticdata:/usr/share/elasticsearch/data/

networks:
  argilla:
    driver: bridge
volumes:
  argilladata:
  elasticdata:
  dbdata:

```

**Comment by tomaarsen** [comment]
After running `build_distribution.sh` and building the updated docker container from this PR locally & using it in your test updated `docker-compose.yaml`, I get the following error on repeat:
```
argilla-argilla-1        | /usr/local/bin/python: No module named argilla.tasks.database.migrate
argilla-argilla-1        | /start_argilla_server.sh: line 6: $'\r': command not found
argilla-argilla-1        | /start_argilla_server.sh: line 20: syntax error: unexpected end of file
argilla-argilla-1 exited with code 2
```

However, if I use your `argilla/argilla-server:pr-2820` as the image instead of my locally built one, then it does work and run seemingly normally, so I probably bunged something up with my local image. Although here the username password seems to be `argilla`/`1234` again...

Long story short, no crashes on my side when using your `argilla/argilla-server:pr-2820`.

File: release.Dockerfile [hunk 2]
```diff
@@ -14,7 +14,11 @@ COPY scripts/start_argilla_server.sh /
 # Copying argilla distribution files
 COPY dist/*.whl /packages/
 
-RUN chmod +x /start_argilla_server.sh && \
+RUN apt-get update && \
+    apt-get install -y python-dev libpq-dev gcc && \
+    pip install psycopg2 && \
+    apt-get remove -y python-dev libpq-dev gcc && \
+    chmod +x /start_argilla_server.sh && \
     for wheel in /packages/*.whl; do pip install "$wheel"[server]; done && \
```
Comment (jfcalvo) [inline release.Dockerfile:22]: Was not possible here to install `postgresql` dependencies?:

`for wheel in /packages/*.whl; do pip install "$wheel"[server,postgresql]; done && \`

With `pip install psycopg2` we are installing an unknown version.

Comment (frascuchon) [reply release.Dockerfile:22]: Okay, I didn't see the optional dep `postgresql`. Yes, we can install from that dep for better dep version control. I will create a separate PR for that. Thanks, @jfcalvo
